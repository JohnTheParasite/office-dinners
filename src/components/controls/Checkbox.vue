<template>
  <div class="input checkbox" :class="{ invalid: hasError, readonly: readonly }">
    <label :for="_uid">
      <input :id="_uid" @change="change" :disabled="readonly" :required="required" v-model="checked" type="checkbox" />
      <span class="checkbox-control">
        <span class="checkbox-check">
          <fa-icon icon="check"></fa-icon>
        </span>
      </span>
      <span class="input-label" v-show="hasLabel" v-html="getLabel"></span>
    </label>
  </div>
</template>

<script>
import control from "./control"
import FaIcon from "@/components/icons/FaIcon"

export default {
  name: "Checkbox",
  components: { FaIcon },
  mixins: [control],
  props: {
    initValue: { default: false }
  },
  data() {
    return {
      checked: false
    }
  },
  computed: {
    value: {
      get() {
        return this.checked
      },
      set(value) {
        this.checked = value
      }
    }
  },
  methods: {
    reset() {
      this.checked = this.initValue
    }
  },
  created() {
    this.reset()
  }
}
</script>

<style scoped lang="scss">
@import "../../scss/components/color";

.checkbox {
  label {
    display: flex;
    flex-direction: row;
    position: relative;
    justify-content: flex-start;
    margin: 0 5px;

    .input-label {
      margin: 0;
      cursor: pointer;
    }

    .checkbox-control {
      margin-right: 5px;
      width: 20px;
      height: 20px;
      border-radius: 2px;
      cursor: pointer;
      transition: all 0.2s ease;
      position: relative;
      display: flex;
      transform: rotate(-90deg);
      align-items: center;
      border: 2px solid rgb(180, 180, 180);
      justify-content: center;

      .checkbox-check {
        display: flex;
        justify-content: center;
        align-items: center;
        width: 100%;
        height: 100%;
        position: absolute;
        left: 0;
        transition: all 0.2s ease;
        z-index: 10;
        transform-origin: right;
        transform: translate(100%);

        i {
          font-size: 11px;
          text-align: center;
          color: inherit;
          transition: all 0.2s ease-out;
          opacity: 0;
          backface-visibility: visible;
          direction: ltr;
        }
      }
    }

    input {
      opacity: 0;
      width: 20px;
      height: 20px;
      position: absolute;

      &:checked + .checkbox-control {
        border: 2px solid rgba($primary, 1) !important;
        background: rgba($primary, 1);
        transform: rotate(0deg);

        .checkbox-check {
          transform: translate(0);

          i {
            opacity: 1;
            color: $white;
            display: flex;
            transform: translate(0);
          }
        }
      }

      &:disabled {
        cursor: default;
        pointer-events: none;

        & + .checkbox-control {
          opacity: 0.4;
          cursor: default;
          pointer-events: none;

          .checkbox-check {
            cursor: default;
            pointer-events: none;
          }
        }
      }
    }
  }

  &.readonly {
    .input-label {
      cursor: default;
    }
  }
}
</style>
