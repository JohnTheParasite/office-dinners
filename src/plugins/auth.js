import AuthService from "@/services/authService";

export const Auth = {
  install(Vue) {
    Vue.prototype.$authService = AuthService;
  },
};