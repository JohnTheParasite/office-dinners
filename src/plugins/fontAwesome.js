import { library } from "@fortawesome/fontawesome-svg-core"
import {
  faArrowRightToBracket,
  faArrowUpRightFromSquare,
  faBan,
  faBurger,
  faChartLine,
  faCheck,
  faChevronDown,
  faChevronLeft,
  faChevronRight,
  faEye,
  faEyeSlash,
  faHandHoldingDollar,
  faHeart,
  faPencil,
  faPercent,
  faTrashCan,
  faUser,
  faUsers,
  faUtensils,
  faXmark
} from "@fortawesome/free-solid-svg-icons"
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome"
import Vue from "vue"

library.add(faChevronLeft)
library.add(faChevronRight)
library.add(faChevronDown)
library.add(faArrowUpRightFromSquare)
library.add(faCheck)
library.add(faPencil)
library.add(faTrashCan)
library.add(faBan)
library.add(faPercent)
library.add(faChartLine)
library.add(faUsers)
library.add(faBurger)
library.add(faHandHoldingDollar)
library.add(faUtensils)
library.add(faHeart)
library.add(faUser)
library.add(faArrowRightToBracket)
library.add(faXmark)
library.add(faEye)
library.add(faEyeSlash)
Vue.component("font-awesome-icon", FontAwesomeIcon)
