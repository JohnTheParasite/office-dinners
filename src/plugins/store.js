import store from "@/store/store"

export const Store = {
  install(Vue) {
    Vue.prototype.$store = store
  }
}
