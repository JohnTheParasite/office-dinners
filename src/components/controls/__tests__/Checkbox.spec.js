import { shallowMount } from "@vue/test-utils"
import Checkbox from "@/components/controls/Checkbox"
import "@/plugins/fontAwesome"

describe("checkbox", () => {
  let checkboxLabel = "test"
  it("should render unchecked checkbox", () => {
    let component = shallowMount(Checkbox)
    expect(component).toMatchSnapshot()
  })

  it("check should works", () => {
    let component = shallowMount(Checkbox)
    expect(component.vm.checked).toBeFalsy()
    component.vm.checked = true
    component.vm.change()
    expect(component.vm.checked).toBeTruthy()
    component.vm.reset()
    expect(component.vm.checked).toBeFalsy()
  })

  it("should render correct label", () => {
    let component = shallowMount(Checkbox, {
      propsData: {
        label: checkboxLabel
      }
    })
    expect(component.vm.getLabel).toBe(checkboxLabel)
  })

  it("should render correct required label", () => {
    let component = shallowMount(Checkbox, {
      propsData: {
        label: checkboxLabel,
        required: true
      }
    })
    expect(component.vm.getLabel).toBe(`<span>${checkboxLabel}<span class="required-star">*</span></span>`)
  })

  it("should set value", () => {
    let component = shallowMount(Checkbox)
    expect(component.vm.checked).toBeFalsy()
    component.vm.value = true
    expect(component.vm.checked).toBeTruthy()
  })
})
