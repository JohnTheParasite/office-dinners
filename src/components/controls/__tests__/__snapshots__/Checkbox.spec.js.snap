// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox should render unchecked checkbox 1`] = `<div class="input checkbox form-group"><label for="2"><input id="2" type="checkbox"> <span class="checkbox-control"><span class="checkbox-check"><font-awesome-icon-stub icon="fa-solid fa-check"></font-awesome-icon-stub></span></span> <span class="input-label" style="display: none;"></span></label></div>`;
